src/promptflow/promptflow/_sdk/entities/_trace.py

```diff
@@ -244,6 +244,12 @@ def _from_spans(spans: typing.List[Span]) -> "LineRun":
             else:
                 # eager flow/arbitrary script
                 main_line_run_data = _LineRunData._from_root_span(span)
+        # main line run span is absent, ignore this line run
+        # this may happen when the line is still executing, or terminated;
+        # or the line run is killed before the traces exported
+        if main_line_run_data is None:
```

Comment:
do we need to log a warning?

Reply:
offline synced, this should be updated with design review